_posts/2024-03-22-dependency-management.md

```diff
@@ -0,0 +1,281 @@
+to extensions, each extension can have different levels of exposure to instability from dependencies. For example, some
+extensions may choose to only depend on highly mature, stable libraries with good portability, whereas others may choose
+to include more experimental dependencies with limited portability. This choice is then forwarded to the user by
+allowing them to choose which extension to use.
+
+At DuckDB, this realization of the importance of extensions and its relation to the no-dependency rule came
+[very early](https://github.com/duckdb/duckdb/pull/594), and consequently extensibility has been ingrained into DuckDB’s
+design from its early days. Today, many parts of DuckDB can be extended. For example, you can add functions (table,
+scalar, copy, aggregation), filesystems, parsers, optimizer rules, and much more. Many new features that are added to
+DuckDB are added in extensions and are grouped by either functionality, or by set of dependencies. Some examples of
+extensions are the [JSON](/docs/extensions/json.html) extension for JSON support or the
+[Spatial](/docs/extensions/spatial.html) extension which offers support for a wide range of geospatial processing
+features. DuckDB’s extensions are distributed as loadable binaries for most major platforms (including
+[DuckDB-Wasm](2023/12/18/duckdb-extensions-in-wasm.html)), allowing loading and installing extension with two simple SQL
+statements:
+
+```sql
+INSTALL json;
+LOAD json;
+```
+
+For most extensions, DuckDB even has an auto-install and auto-load feature which will detect the required extensions for
+a SQL statement and automatically install and load them. For a detailed description on which extensions are available
+and how to use them, check out the [docs](https://duckdb.org/docs/extensions/overview.html).
+
+## Dependency management
+So far, we’ve seen how DuckDB avoids external dependencies in its core codebase by pushing them out of core into
+extensions. However, we’re not out of the woods yet. As DuckDB is written in C++, the most natural way to write
+extensions is C++. In C++ though, there is no standard tooling like a package manager and the answer to the
+question of how to do dependency management in C++ has been, for many years: *“Through much pain and anguish.”* Given 
+DuckDB’s focus on portability and support for many platforms, managing dependencies manually is not feasible: dependencies generally are built from source, with each their own intricacies requiring special build flags and
+configuration for different platforms. With a growing ecosystem of extensions, this would quickly turn into an
+unmaintainable mess.
+
+Fortunately, much has changed in the C++ landscape over the past few years. Today, good dependency managers do exist.
+One of them is Microsoft’s [vcpkg](https://vcpkg.io/). It has arguably become the _de facto_ standard in C++ dependency
+managers, as proven by its 20k+ GitHub stars and native support
```

Comment:
```
It has arguably become the _de facto_ standard in C++ dependency managers, as proven by its 20k+ GitHub stars and native support
```
I think this is going a bit too far? I would maybe turn this around and saying that while you were tasked with maintaining extensions for duckdb (that also gives validation on why you are talking about this), you found that whatever dependecy you seemed to need were covered by vcpkg, experimented with it, found it it was solving a lot of hassle, and it has been relied on for now a year.

Reply:
yea @Mytherin also did not agree with this sentence. However, I feel like I give enough arguments to make the statement. And i also include the word `arguably` to indicate that its up for debate whether the statement is true.

Reply:
ok nvm i will tone it down